test/unit/component/registry/bpmn-model-filters.test.ts

```diff
@@ -518,4 +518,47 @@ describe('Bpmn Model filters', () => {
       bpmnElementKind: ShapeBpmnElementKind.POOL,
     });
   });
+
+  it('Filter a black pool (No processRef)', () => {
+    const originalBpmnModel = toBpmnModel({
+      pools: {
+        id: 'participant_id_1',
+        name: 'Participant 1',
+      },
+    });
+
+    expect(modelFiltering.filter(originalBpmnModel, { pools: { id: 'participant_id_1' } })).toStrictEqual(originalBpmnModel);
+  });
+
+  it('Filter a NOT displayed pool containing BPMN elements', () => {
+    const originalBpmnModel = toBpmnModel({
+      pools: {
+        id: 'participant_id_1',
+        name: 'Participant 1',
+        hidden: true,
+        tasks: {
+          id: 'task_1',
+          name: 'Task 1',
+        },
+        startEvents: {
+          id: 'start_event_1',
+        },
+        sequenceFlows: {
+          id: 'sequence_flow_1',
+          source: 'start_event_1',
+          target: 'task_1',
+        },
+        lanes: {
+          id: 'lane_1',
+        },
+      },
+    });
+
+    expect(modelFiltering.filter(originalBpmnModel, { pools: { id: 'participant_id_1' } })).toStrictEqual({
+      pools: [],
+      lanes: originalBpmnModel.lanes,
+      flowNodes: originalBpmnModel.flowNodes,
+      edges: originalBpmnModel.edges,
+    });
+  });
```

Comment:
We don't know what we are testing
check model + add test for model builder: see https://github.com/process-analytics/bpmn-visualization-js/pull/2129#discussion_r930018616

Reply:
we will do it in another PR